python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -690,11 +689,17 @@ def _flush(self, objs):
             if not obj._dirty:
                 continue
 
+            if hasattr(obj, _reg_id_str):
+                obj_id = getattr(obj, _reg_id_str)
+                if obj_id not in self._loaded_ids:
+                    continue
+            else:
+                continue
```

Comment:
I haven't tested whether this is the case (I can if we're really paranoid about this) but the ```_dirty``` flag would be set if the object is modified at all regardless of if the 3rd modification causes an exception to be thrown (but the object to still exist in the list of objects known to the repository).
I plan to add a line which sets the dirty status of jobs freshly loaded from disk to be not dirty after the loading has finished but atm we currently don't have this either.